storage/series/reader.go

```diff
@@ -126,7 +126,9 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 					return nil, err
 				}
 				if stream != nil {
-					results = append(results, []xio.SegmentReader{stream})
+					results = append(results, []xio.BlockReader{
+						xio.NewBlockReader(stream, start, end),
```

Comment:
The block start and end time though is not the query's start and end time, it's different per block.

Can you just perhaps make the block interface return a `BlockReader` which would make more sense here?